src/Components/Server/src/DependencyInjection/ComponentServiceCollectionExtensions.cs

```diff
@@ -33,6 +34,9 @@ public static IServerSideBlazorBuilder AddServerSideBlazor(this IServiceCollecti
 
             services.AddDataProtection();
 
+            services.TryAddScoped<ProtectedLocalStorage>();
+            services.TryAddScoped<ProtectedSessionStorage>();
```

Comment:
This surprises me a bit. Do we definitely want to include the protected storage services by default?

For non-core functionality like this it seems more idiomatic for the developer to call `services.AddProtectedBrowserStorage()` (or similar) themselves to opt in.

I suppose that is a bit of an arbitrary hoop we're giving them to jump through, but it does mean they are making more of a conscious choice to plug it into their application.

Perhaps this is a question for API review.

Reply:
> but it does mean they are making more of a conscious choice to plug it into their application.

* Users are making that choice by using these types. Having the service registration isn't a significant cost and we have precedence with MVC registering a whole bunch of optional features \ services in DI behind a single gesture. 

* Previously using these APIs required bringing in an optional dependency (DataProtection), which is no longer the case. I would have felt more strongly about a separate API if this required bringing in more optional services.

* Generally our service registration chain in dependencies. In this case, `AddProtectedBrowserStorage` would only make sense if you had already called `AddServerSideBlazor`. That would require us to either call  `AddServerSideBlazor` on your behalf or use a builder pattern, both of which seem unnecessary.

Reply:
The difference is that libraries can start using this without the app developer knowing. I'm not sure if that's something we want, given that this will put data in the browser (even though is protected).

I can imagine it being hard to detect if an app brings in a library that starts making use of this "liberally".

Reply:
OK, I see the arguments both ways and am fine with either. I think the developer experience is OK either way.

@pranavkm @javiercn If the two of you can come to an agreement on it let's go with that.

Reply:
I'll let @pranavkm decide. I don't feel strongly either.